src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -153,22 +125,11 @@ public void SetOptions(OptionSet optionSet)
 
                     _currentValues = _currentValues.SetItem(optionKey, setValue);
 
-                    ImmutableArray<Lazy<IOptionSerializer, OptionSerializerMetadata>> optionSerializers;
-                    if (_featureNameToOptionSerializers.TryGetValue(optionKey.Option.Feature, out optionSerializers))
+                    foreach (var serializer in _optionSerializers)
                     {
-                        foreach (var serializer in optionSerializers)
+                        if (serializer.Value.TryPersist(optionKey, setValue))
```

Comment:
this still seems not 100% clear what serializer will take care of which options. can 2 IOptionSerializers serialize same option?